File: 0733-flood-fill/0733-flood-fill.py
class Solution:
    org_color = None
    image = None
    def floodFill(self, image: List[List[int]], sr: int, sc: int, color: int) -> List[List[int]]:
        
        self.org_color = image[sr][sc]
        self.image = image
        
        if self.image[sr][sc]!=color:
            return self.fill(sr,sc, color)
        else:
            return self.image
        
    
    def fill(self, sr: int, sc: int, color:int):
        self.image[sr][sc] = color
            
        neighbors = [(sr+1,sc), (sr-1,sc), (sr,sc+1), (sr, sc-1)]
        
        for neighbor in neighbors:
            if neighbor[0]>= 0 and neighbor[0] < len(self.image) and neighbor[1] >=0  and neighbor[1] < len(self.image[0]):
                if self.image[neighbor[0]][neighbor[1]] == self.org_color:
                    self.fill(neighbor[0], neighbor[1], color)
   
        
        return self.image
        
        

File: 0733-flood-fill/0733-flood-fill.cs
public class Solution {
    public int[][] FloodFill(int[][] image, int sr, int sc, int color) {

        int orig = image[sr][sc];
        if (orig == color) return image;
        Queue<(int x, int y)> q = new Queue<(int, int)>();
        q.Enqueue((sr, sc));
        int m = image.Length;
        int n = image[0].Length;
        image[sr][sc] = color;

        int[,] directions = new int[4,2] {{0,1},{0,-1},{1,0},{-1,0}};

        while(q.Count > 0){
            (int x, int y) curr =  q.Dequeue();

            for (int i = 0; i < 4; i++){
                int newR = curr.x + directions[i, 0];
                int newC = curr.y + directions[i, 1];

                if (0 <= newR && newR < m && 0 <= newC && newC < n && image[newR][newC] == orig){
                    image[newR][newC] = color; 
                     q.Enqueue((newR, newC));
                }
            }
        }

        return image;
    }
}